types/handler.go

```diff
@@ -40,14 +40,18 @@ func ChainAnteDecorators(chain ...AnteDecorator) AnteHandler {
 		return nil
 	}
 
-	// handle non-terminated decorators chain
-	if (chain[len(chain)-1] != Terminator{}) {
```

Comment:
Even though we don't use Terminator anymore, I didn't remove it to not break API compatibility with existing users.

Also I choose to use an AnteHandler implementation instead of an AnteDecorator to reduce the call depth by one for invocations that make it to the terminator.